dali/operators/generic/slice/slice_attr.h

```diff
@@ -246,6 +242,42 @@ class PositionalSliceAttr {
   }
 
  private:
+  template <typename T>
+  span<T> sample_as_span(const TensorListView<StorageCPU, T, DynamicDimensions>& v,
+                         int sample_idx) {
+    ptrdiff_t vol = volume(v.tensor_shape_span(sample_idx));
+    return span<T>(v.tensor_data(sample_idx), vol);
+  }
+
+  template <typename T, typename Backend>
+  std::pair<TensorListView<StorageCPU, T, DynamicDimensions>,
+            TensorListView<StorageCPU, T, DynamicDimensions>>
```

Comment:
Why pair? This function looks like it tries too hard to aggregate stuff, losing the semantics in the process. Now we only have "first, second" instead of "shape, anchor" (or was it "anchor, shape"?).
The previous version, if a bit awkward, was cleaner.